### PR TITLE
Show macOS arm64 architecture in doctor and devices list

### LINKED ISSUE #65976: [doctor] Add macOS architecture to doctor output
Now that there are 2 architectures for macOS (x64 and arm64) we should include that information in the `flutter doctor` output.  We often ask people reporting errors to add `flutter doctor` output and this can help us find bugs related to one architecture.

cc @jonahwilliams @jmagman

### PR DESCRIPTION
**Pull request by jmagman**
## Description

Add `targetPlatformDisplayName` for `flutter doctor` and `flutter devices` to show when the host is an Apple Silicon macOS ARM machine.

Do not add a more substantive `TargetPlatform.darwin_arm64` change since the artifacts, build system, etc all require `darwin_x64` even on ARM to leverage Rosetta translation.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/65976

## Tests
macos_device_tests

### CODE REVIEW

File: packages/flutter_tools/test/src/context.dart [hunk 1]
```diff
@@ -291,11 +291,13 @@ class MockSimControl extends Mock implements SimControl {
 }
 
 class FakeOperatingSystemUtils implements OperatingSystemUtils {
+  FakeOperatingSystemUtils({this.hostPlatform = HostPlatform.linux_x64});
```
Comment (christopherfujino) [inline packages/flutter_tools/test/src/context.dart:294]: Just curious why this change?

**Review by christopherfujino — APPROVED** [review]
LGTM: with a nit question

File: packages/flutter_tools/test/src/context.dart (hunk 1 above)
Comment (jmagman) [reply packages/flutter_tools/test/src/context.dart:294]: There used to be no way to set `hostPlatform` to anything other than `HostPlatform.linux_x64`.  But in my new test I want to set it to `HostPlatform.darwin_x64` and `HostPlatform.darwin_arm`.

Comment (jmagman) [reply packages/flutter_tools/test/src/context.dart:294]: Nope nope nope I misread this code you're right, it had a setter.  Reverting...
